services/frontend/src/components/Search.vue

```diff
@@ -97,21 +112,66 @@ export default defineComponent({
           }
           break;
         }
+        case "Enter": {
+          addPackage();
+          break;
+        }
+        case "Escape": {
+          if (event.target !== null && event.target instanceof HTMLElement) {
+            event.target.blur();
+          }
+          break;
+        }
       }
 
-      selected.value = results[index];
-    }
-
-    function addPackage() {
-      showPackageBuildSelection.value = true;
-      showResults.value = false;
+      if (!(index > results.length || index < 0)) {
+        selected.value = results[index];
+      }
     }
 
     function onInput() {
-      search(term.value).then(resp => {
+      const regexmatch = gitUrlRegex.exec(term.value);
+      if (regexmatch !== null) {
+        const res = NewResult();
+
+        res.ID = -1;
+        res.URL = term.value;
+
+        const urlpath = regexmatch[2].split("/");
+        res.Name = urlpath[urlpath.length - 1];
+
         results.splice(0, results.length);
-        results.push(...resp);
-      });
+        results.push(res);
+      } else if (
+        ["git://", "ssh://", "http://", "https://"].some(i => term.value.startsWith(i)) ||
```

Comment:
`includes("/")` will catch everything the `startsWith` catches as well, so it seems redundant